Automattic/wp-calypso
# Stepper: Extend Link in Bio flow

## What
This PR makes Stepper capable of extending a flow. In our case we're extending Link In Bio flow to create a Link In Bio Domain flow.

This flow needs to:
1. Register / Login
2. Choose a design (from the link-in-bio themes)
3. Choose the profile image, site name, and description
4. Choose the plan (personal or premium)
5. Checkout with the selected plan and the domain connection in the cart
6. Immediately after checkout, the user will be redirected to the domain connection setup flow: `https://wordpress.com/domains/mapping/[sitename]/setup/[domain name]?firstVisit=true`

## Testing
1. Checkout and `yarn start` ( Or Calypso live link)
2. Use `calypso.localhost:3000/setup/link-in-bio-domain?domain=yourdomain.com&provider=test-provider`
3. Check that the flow works, and that the events get fired with domain and provider 

Related to https://github.com/Automattic/wp-calypso/issues/75048
